samtranslator/translator/arn_generator.py

```diff
@@ -3,12 +3,18 @@
 class ArnGenerator(object):
 
     @classmethod
-    def generate_arn(cls, partition, service, resource):
+    def generate_arn(cls, partition, service, resource, include_account_id=True):
         if not service or not resource:
             raise RuntimeError("Could not construct ARN for resource.")
 
-        return 'arn:{0}:{1}:${{AWS::Region}}:${{AWS::AccountId}}:{2}'.format(partition, service, resource)
+        arn = 'arn:{0}:{1}:${{AWS::Region}}:'
 
+        if include_account_id:
+            arn += '${{AWS::AccountId}}:'
+
+        arn += '{2}'
```

Comment:
If we don't add an accountID we'd still have to add the corresponding colon and asterix or we'd evaluate the resource as the account id